### PR TITLE
cleanup: Use single expectation for each It block

### PR DESCRIPTION
**Pull request by xtqqczze**
# PR Summary

## PR Context

Convention is to assert a single expectation for each `It` block. [[1]](https://github.com/pester/Pester/wiki/It)

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` or `[ WIP ]` to the beginning of the title (the `WIP` bot will keep its status check at `Pending` while the prefix is present) and remove the prefix when the PR is ready.
- **[Breaking changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)**
    - [x] None
    - **OR**
    - [ ] [Experimental feature(s) needed](https://github.com/MicrosoftDocs/PowerShell-Docs/blob/staging/reference/6/Microsoft.PowerShell.Core/About/about_Experimental_Features.md)
        - [ ] Experimental feature name(s): 
- **User-facing changes**
    - [x] Not Applicable
    - **OR**
    - [ ] [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [ ] Issue filed: 
- **Testing - New and feature**
    - [x] N/A or can only be tested interactively
    - **OR**
    - [ ] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
- **Tooling**
    - [ ] I have considered the user experience from a tooling perspective and don't believe tooling will be impacted.
    - **OR**
    - [x] I have considered the user experience from a tooling perspective and enumerated concerns in the summary. This may include:
        - Impact on [PowerShell Editor Services](https://github.com/PowerShell/PowerShellEditorServices) which is used in the [PowerShell extension](https://github.com/PowerShell/vscode-powershell) for VSCode (which runs in a different PS Host).
        - Impact on Completions (both in the console and in editors) - one of PowerShell's most powerful features.
        - Impact on [PSScriptAnalyzer](https://github.com/PowerShell/PSScriptAnalyzer) (which provides linting & formatting in the editor extensions).
        - Impact on [EditorSyntax](https://github.com/PowerShell/EditorSyntax) (which provides syntax highlighting with in VSCode, GitHub, and many other editors).

### CODE REVIEW

File: test/powershell/Modules/ThreadJob/ThreadJob.Tests.ps1 [hunk 1]
```diff
@@ -93,24 +93,30 @@ Describe 'Basic ThreadJob Tests' -Tags 'CI' {
 
         $job = Start-ThreadJob -ScriptBlock { "Goodbye" } -InitializationScript { "Hello" }
         $results = $job | Receive-Job -Wait
-        $results[0] | Should -Be "Hello"
-        $results[1] | Should -Be "Goodbye"
+        $results | Should -BeExactly (
+            "Hello",
+            "Goodbye"
+        )
```
Comment (vexx32) [inline test/powershell/Modules/ThreadJob/ThreadJob.Tests.ps1:99]: You can drop the commas on all of these if you use `@()`:
```suggestion
        $results | Should -BeExactly @(
            "Hello"
            "Goodbye"
        )
```

File: test/powershell/Modules/ThreadJob/ThreadJob.Tests.ps1 [hunk 2]
```diff
@@ -290,7 +304,7 @@ Describe 'Basic ThreadJob Tests' -Tags 'CI' {
 '@
 
         $result = & "$PSHOME/pwsh" -c $script
-        $result | Should -BeExactly "True"
+        $result | Should -BeTrue
```
Comment (iSazonov) [inline test/powershell/Modules/ThreadJob/ThreadJob.Tests.ps1:307]: $result contains a string from an external application.
Please revert the change.

File: test/powershell/Modules/ThreadJob/ThreadJob.Tests.ps1 [hunk 3]
```diff
@@ -327,7 +341,7 @@ Describe 'Basic ThreadJob Tests' -Tags 'CI' {
 '@
 
         $result = & "$PSHOME/pwsh" -c $script
-        $result | Should -BeExactly "True","True","True"
+        $result | Should -BeTrue
```
Comment (iSazonov) [inline test/powershell/Modules/ThreadJob/ThreadJob.Tests.ps1:344]: The same.

File: test/powershell/Modules/ThreadJob/ThreadJob.Tests.ps1 (hunk 1 above)
Comment (iSazonov) [reply test/powershell/Modules/ThreadJob/ThreadJob.Tests.ps1:99]: It makes the test less readable. I can not accept this.

**Review by iSazonov — CHANGES_REQUESTED** [review]
I can not accept these changes and suggest to close the PR.

File: test/powershell/Modules/ThreadJob/ThreadJob.Tests.ps1 [hunk 4]
```diff
@@ -130,12 +136,14 @@ Describe 'Basic ThreadJob Tests' -Tags 'CI' {
...
-        $results[4] | Should -Be $Var4
-        $results[5] | Should -Be $global:GVar1
+        $results | Should -BeExactly @(
+            $Var1
+            $Var2
+            $Var3
+            2
+            $Var4
+            $global:GVar1
+        )
```
Comment (iSazonov) [inline test/powershell/Modules/ThreadJob/ThreadJob.Tests.ps1:146]: I think it is not correct. I can not approve such change.

I suggest close the PR.

Comment (xtqqczze) [reply test/powershell/Modules/ThreadJob/ThreadJob.Tests.ps1:146]: `Expected exactly @('Hello', 'Goodbye', 102, 2, 1, 2, 3, 4, 5, 'GlobalVar'), but got @('Hello', 'Goodbye', 102, 2, @(1, 2, 3, 4, 5), 'GlobalVar').`

Comment (iSazonov) [reply test/powershell/Modules/ThreadJob/ThreadJob.Tests.ps1:146]: I found issues in Pester repository about arrays processing. So I'd don't want to spend time to investigate this because this does not add many value to our tests. I suggest close the PR and continue with others PRs.

Comment (vexx32) [reply test/powershell/Modules/ThreadJob/ThreadJob.Tests.ps1]: Array processing is typically only an issue when dealing with strongly-typed collections (e.g., KeyValueCollection) which aren't quite the same / don't implement all the same interfaces.

This specific example will fail due to pipeline unrolling of arrays; this should work here:
```ps1
        $results | Should -BeExactly @(
            $Var1
            $Var2
            $Var3
            2
            ,$Var4
            $global:GVar1
        )
```

**Comment by iSazonov** [comment]
The PR looks very dangerous. I believe this does not add benefits. I close.
